src/mlpack/methods/ann/gan/gan.hpp

```diff
@@ -76,10 +76,7 @@ class GAN
    * @param clippingParameter Weight range for enforcing Lipschitz constraint.
    * @param lambda Parameter for setting the gradient penalty.
    */
-  GAN(arma::mat& trainData,
-      Model generator,
-      Model discriminator,
-      InitializationRuleType& initializeRule,
+  GAN(InitializationRuleType& initializeRule,
```

Comment:
Not sure I see the reason to move the generator into the ``ResetData`` function, I think it would be great if a user could pass the generator.

Reply:
Ah, yes earlier there was reason but it would have produced some error. So I have shited `generator` and `discriminator` back to constructor.
Earlier I was thinking that a user will not need to provide `generator` and `discriminator` during loading so thought it's better not to include them during `object creation`. 
But it would have produced errors because a user may call `gan::predict()` before any `training` and that would have raised errors as `generator` and `discriminator` networks were absent.
So I have moved them back to constructor.

Reply:
Thanks for the clarification.